src/core/mesh/qgsmesheditor.h

```diff
@@ -0,0 +1,264 @@
+/***************************************************************************
+  qgsmesheditor.h - QgsMeshEditor
+
+ ---------------------
+ begin                : 8.6.2021
+ copyright            : (C) 2021 by Vincent Cloarec
+ email                : vcloarec at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSMESHEDITOR_H
+#define QGSMESHEDITOR_H
+
+#include <QObject>
+#include <QUndoCommand>
+#include <QPointer>
+
+#include "qgsmeshdataset.h"
+#include "qgsmeshdataprovider.h"
+#include "qgstriangularmesh.h"
+#include "qgstopologicalmesh.h"
+
+/**
+ * \ingroup core
+ *
+ * \brief Class that represents an error during mesh editing
+ *
+ * \since QGIS 3.22
+ */
+class CORE_EXPORT QgsMeshEditingError
+{
+  public:
+
+    //! Type of the error
+    enum MeshEditingErrorType
```

Comment:
Can you please move this to the Qgis class, and promote it to an enum class (also add docs for the individual member values)? We're moving toward having all enums in a common class (Qt style) to easily facilitate forward declaration vs header imports.

Reply:
done